options/options.go

```diff
@@ -0,0 +1,217 @@
+	SetSliceMapNilIfLenZero bool
+	// SetSliceMapRandomToZero sets random integer generation to zero for slice and maps
+	SetSliceMapRandomToZero bool
+	// RandomIntegerBoundary sets boundary random integer value generation. Boundaries can not exceed integer(4 byte...)
+	RandomIntegerBoundary *interfaces.RandomIntegerBoundary
+	// RandomFloatBoundary sets the boundary for random float value generation. Boundaries should comply with float values constraints (IEEE 754)
+	RandomFloatBoundary *interfaces.RandomFloatBoundary
+}
+
+// MaxDepthOption used for configuring the max depth of nested struct for faker
+type MaxDepthOption struct {
+	typeSeen          map[reflect.Type]int
+	recursionMaxDepth int
+}
+
+func (o *MaxDepthOption) RememberType(t reflect.Type) {
+	o.typeSeen[t]++
+}
+
+func (o *MaxDepthOption) ForgetType(t reflect.Type) {
+	o.typeSeen[t]--
+}
+
+func (o *MaxDepthOption) RecursionOutOfLimit(t reflect.Type) bool {
+	return o.typeSeen[t] > o.recursionMaxDepth
+}
+
+// BuildOptions build all option functions into one option
+func BuildOptions(optFuncs []OptionFunc) *Options {
+	ops := DefaultOption()
+
+	for _, optFunc := range optFuncs {
+		optFunc(ops)
+	}
+
+	return ops
+}
+
+// DefaultOption build the default option
+func DefaultOption() *Options {
+	ops := &Options{}
+	ops.MaxDepthOption = &MaxDepthOption{
+		typeSeen:          make(map[reflect.Type]int, 1),
+		recursionMaxDepth: 1,
+	}
+	ops.StringLanguage = &interfaces.LangENG
+	ops.RandomStringLength = 25            //default
+	ops.RandomMaxSliceSize = 100           //default
+	ops.RandomMinSliceSize = 0             // default
+	ops.MaxGenerateStringRetries = 1000000 //default
+	ops.RandomIntegerBoundary = &interfaces.DefaultIntBoundary
+	ops.RandomFloatBoundary = &interfaces.DefaultFloatBoundary
+	return ops
+}
+
+// OptionFunc define the options contract
+type OptionFunc func(oo *Options)
+
+// WithFieldsToIgnore used for ignoring a field when generating the fake data
+func WithFieldsToIgnore(fieldNames ...string) OptionFunc {
+	return func(oo *Options) {
+		if oo.IgnoreFields == nil {
+			oo.IgnoreFields = make(map[string]struct{}, len(fieldNames))
+		}
+		for _, f := range fieldNames {
+			oo.IgnoreFields[f] = struct{}{}
+		}
+	}
+}
+
+// WithCustomFieldProvider used for storing the custom provider function
+func WithCustomFieldProvider(fieldName string, provider interfaces.CustomProviderFunction) OptionFunc {
+	return func(oo *Options) {
+		if oo.FieldProviders == nil {
+			oo.FieldProviders = make(map[string]interfaces.CustomProviderFunction, 1)
+		}
+		oo.FieldProviders[fieldName] = provider
+	}
+}
+
+// WithRecursionMaxDepth used for configuring the max depth of nested struct for faker
+func WithRecursionMaxDepth(depth uint) OptionFunc {
+	return func(oo *Options) {
+		if oo.MaxDepthOption == nil {
+			oo.MaxDepthOption = &MaxDepthOption{
+				recursionMaxDepth: 1, // default
+				typeSeen:          make(map[reflect.Type]int, 1),
+			}
+		}
+		if depth > 0 {
```

Comment:
I think `depth` can set 0, which means disabling recursion.

Reply:
Alright, so the default valuer is `0` is that correct?
I will address all the comment tomorrow weekend

Reply:
Not really, allow recursive one times is more suitable for my use case